crates/matrix-sdk-ui/src/timeline/builder.rs

```diff
@@ -393,6 +393,10 @@ async fn room_event_cache_updates_task(
                         },
                     )
                     .await;
+
+                if matches!(origin, EventsOrigin::Cache) {
+                    timeline_controller.retry_event_decryption(None).await;
+                }
```

Comment:
Hmm, as far as I understand, this gets called once per event, while `retry_event_decryption(None)` tries to decrypt all the events in the timeline.

I wouldn't be surprised if that turns out to be a problem.

We should really move this decryption logic ASAP.

Reply:
It is called per batch of updates, so it can cover multiple events.

Reply:
Ah ok, all good then.